### ISSUE TITLE
Prometheus: label filters - use /label match param 

### ISSUE THREAD

**Issue by gtk-grafana** [grafana/grafana#58009]
While we added support to use the Prometheus label match parameter in https://github.com/grafana/grafana/pull/57553 for the list of metrics, this didn't update the label filters endpoint, which is still calling the /series endpoint.

What I have noticed is first call (under Metric) is making a `resources/api/v1/label/__name__/values` call. But very next `Label Filters` calls are again `/resources/api/v1/series` 
Can we have `label` calls for Label filters as well?
<img width="615" alt="image" src="https://user-images.githubusercontent.com/960211/199303836-b53b94cd-e8aa-45f5-a142-928f6a0d8ef5.png">

_Originally posted by @adigov in https://github.com/grafana/grafana/issues/57553#issuecomment-1298899555_

**Comment by gtk-grafana** [comment]
After taking a quick look at this, I'm not sure this is something we can accomplish with the /labels endpoint. https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/language_provider.ts#L506 The response of the /series endpoint uses more then just the labels, it also consumes the values of the labels.

For example, the current /series endpoint:

GET: `http://localhost:3000/api/datasources/5/resources/api/v1/series?start=1667323769&end=1667327369&match%5B%5D=%7B__name__%3D%22access_permissions_duration_bucket%22%7D`
```
{"status":"success","data":[{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"+Inf"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"0.00016"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"0.00064"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"0.00256"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"0.01024"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"0.04096"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"0.16384"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"0.65536"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"1e-05"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"2.62144"},{"__name__":"access_permissions_duration_bucket","instance":"host.docker.internal:3000","job":"grafana","le":"4e-05"}]}
```

Which is processed by `fetchSeriesLabels` into an object like:
```
{
    "instance": [
        "host.docker.internal:3000"
    ],
    "job": [
        "grafana"
    ],
    "le": [
        "+Inf",
        "0.00016",
        "0.00064",
        "0.00256",
        "0.01024",
        "0.04096",
        "0.16384",
        "0.65536",
        "1e-05",
        "2.62144",
        "4e-05"
    ]
}
```

However if we query the /labels endpoint with that same query:
We'll get back: `{"status":"success","data":["__name__","instance","job","le"]}`

TL;DR We use the results of the series endpoint to populate the potential values of those labels in the current query builder.
While we could potentially change the functionality of the query builder to use the new labels endpoint, it would require that we fire another API call to get the values after the user has selected an individual metric label.

This could be a potential avenue for additional optimization especially if we want to support thousands to millions of labels for a specific metric.

I'm not going to close this yet as I'm curious if this is something people want, but giving a low priority for now.

**Comment by crsandeep** [comment]
@gtk-grafana mainly for performance reason. This is what I observed in our prometheus deployment

`/api/v1/series?match[]=up&start=1667365242&end=1667368842` fails to execute since it times out after a min. We have huge amount of time series matching `up`

However

`/api/v1/labels?match[]=up&start=1667365242&end=1667368842` loads under a second for me
`/api/v1/label/pool/values?match[]=up&start=1667365242&end=1667368842` the subsequent call to get the values of the label I selected took about 2 second

**Comment by gtk-grafana** [comment]
Merged in 9.2.5! Thank you @crsandeep and @adigov for your help on this!
